pkg/clusterapi/etcd.go

```diff
@@ -0,0 +1,86 @@
+package clusterapi
+
+import (
+	etcdbootstrapv1 "github.com/aws/etcdadm-bootstrap-provider/api/v1beta1"
+	etcdv1 "github.com/aws/etcdadm-controller/api/v1beta1"
+	v1 "k8s.io/api/core/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	bootstrapv1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1beta1"
+	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
+
+	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/constants"
+)
+
+// SetEtcdConfigInEtcdCluster sets up the etcd config in EtcdadmCluster.
+func SetEtcdConfigInEtcdCluster(etcd *etcdv1.EtcdadmCluster, version string) {
+	etcd.Spec.EtcdadmConfigSpec.CloudInitConfig = &etcdbootstrapv1.CloudInitConfig{
+		Version:    version,
+		InstallDir: "/usr/bin",
+	}
+	etcd.Spec.EtcdadmConfigSpec.PreEtcdadmCommands = []string{
+		"hostname \"{{`{{ ds.meta_data.hostname }}`}}",
+		"echo \"::1         ipv6-localhost ipv6-loopback\" >/etc/hosts",
+		"echo \"127.0.0.1   localhost\" >>/etc/hosts",
+		"echo \"127.0.0.1   {{`{{ ds.meta_data.hostname }}`}}\" >>/etc/hosts",
+		"echo \"{{`{{ ds.meta_data.hostname }}`}}\" >/etc/hostname",
+	}
+}
+
+// SetEtcdConfigInCluster sets up the etcd config in CAPI Cluster.
+func SetEtcdConfigInCluster(cluster *clusterv1.Cluster, clusterSpec *cluster.Spec) {
```

Comment:
why not pass the `EtcdadmCluster` here to set the ref? I believe we do that with other objects that are referenced